pymeasure/instruments/anritsu/anritsuMS464xB.py

```diff
@@ -0,0 +1,318 @@
+        """Gives Enum member by specifying name or value.
+
+        Args:
+            input_value (str|int): Enum name or value
+
+        Returns:
+            _type_: enum member
+        """
+
+        if isinstance(input_value, int):
+            return cls(input_value)
+        else:
+            return cls[input_value.upper()]
+
+
+class VNAChannel(Channel):
+    """Implementation of a Anritsu MS464xB channel."""
+
+    _SWEEP_TYPES = {"LIN", "LOG", "FSEGM", "ISEGM", "POW", "MFGC"}
+
+    center_frequency = Instrument.control(
+        ":FREQ:CENT?",
+        ":FREQ:CENT %d",
+        """Center Frequency in Hz.""",
+        validator=strict_range,
+        values=[30000, 8000000000],
+    )
+
+    cw_frequency = Instrument.control(
+        ":FREQ:CW?",
+        ":FREQ:CW %d",
+        """CQ Frequency in Hz.""",
+        validator=strict_range,
+        values=[30000, 8000000000],
+    )
+
+    span_frequency = Instrument.control(
+        ":FREQ:SPAN?",
+        ":FREQ:SPAN %d",
+        """Span Frequency in Hz""",
+        validator=strict_range,
+        values=[1, 8000000000],
+    )
+
+    start_frequency = Instrument.control(
+        ":FREQ:STAR?",
+        ":FREQ:STAR %d",
+        """ Starting frequency in Hz.""",
+        validator=strict_range,
+        values=[1, 8000000000],
+    )
+
+    stop_frequency = Instrument.control(
+        ":FREQ:STOP?",
+        ":FREQ:STOP %d",
+        """ Stop frequency in Hz.""",
+        validator=strict_range,
+        values=[1, 8000000000],
+    )
+
+    sweep_type = Instrument.control(
+        ":SWE:TYP?",
+        ":SWE:TYP %s",
+        """ Sweep Type. Valid options are LIN|LOG|FSEGM|ISEGM|POW|MFGC.""",
+        validator=strict_discrete_set,
+        values=_SWEEP_TYPES,
+    )
+
+    sweep_mode = Instrument.control(
+        ":SA:MODE?",
+        ":SA:MODE %s",
+        """ Sweep mode. Valid options are VNA|CLAS.""",
+        validator=strict_discrete_set,
+        values=["VNA", "CLAS"],
+    )
+
+    sweep_points = Instrument.control(
+        ":SWE:POI?", ":SWE:POI %d", """ Sweep point.""", validator=strict_range, values=[2, 100000]
+    )
+
+    num_averages = Instrument.control(
+        ":AVER:COUN?",
+        ":AVER:COUN %d",
+        """ Sets the number of averages.""",
+        validator=strict_range,
+        values=[1, 20000],
+    )
+
+    bandwidth = Instrument.control(
+        ":BWID?",
+        ":BWID %d",
+        """ Sets the IF bandwidth.""",
+        validator=strict_range,
+        values=[1, 1000000],
+    )
+
+    def insert_id(self, command):
+        return f":SENS{self.id}{command}"
+
+
+class AnritsuMS464xB(Instrument):
+    """Represents the Anritsu MS474xB VNA. Implements controls to change the analysis
+    range and to retreve the data for the trace.
+    """
+
+    channels = Instrument.ChannelCreator(
+        VNAChannel, (1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16)
+    )
+
+    class BinaryData_Byte_Format(CustomEnum):
```

Comment:
```suggestion
    class BinaryDataByteFormat(CustomEnum):
```
to be consistent with the other classes' use of CamelCase?